src/briefcase/commands/build.py

```diff
@@ -14,16 +17,66 @@ def add_options(self, parser):
         self._add_update_options(parser, context_label=" before building")
         self._add_test_options(parser, context_label="Build")
 
-    def build_app(self, app: BaseConfig, **options):
+    def build_app(self, app: AppConfig, **options):
         """Build an application.
 
         :param app: The application to build
         """
         # Default implementation; nothing to build.
 
+    def verify_template(self, app: AppConfig):
+        """Verify the template satisfies the Command's requirements."""
+        self.logger.info("Checking template compatability...", prefix=app.app_name)
+
+        with self.input.wait_bar("Verifying Briefcase support..."):
+            # check against the "release" for each Version since versions
+            # such as "0.3.15.dev123" are considered older than "0.3.15"
+
+            if template_target := self.briefcase_target_version(app=app):
+                template_target = Version(template_target).release
+
+            if platform_target := self.oldest_compatible_briefcase:
+                platform_target = Version(platform_target).release
+
+            if platform_target and not template_target:
+                raise BriefcaseCommandError(
+                    f"""\
+The app template must declare a target version of Briefcase to confirm it is
+compatible with this version of Briefcase.
+
+The template's target Briefcase version must be {'.'.join(map(str, platform_target))} or later.
+
+Once the template is updated, run the create command:
+
+    $ briefcase create {self.platform} {self.output_format}
```

Comment:
Can't verify this renders 100% correctly, but suggested text:
```suggestion
Briefcase requires that the app template explicitly declare that it is compatible with 
Briefcase {'.'.join(map(str, platform_target))} or later. However, the generated app's 
`briefcase.toml` has no `target_version` declaration. 

If the app was generated with an earlier version of Briefcase using the default Briefcase
template, you can run:

     $ briefcase create {self.platform} {self.output_format}

to re-generate your app. 

If you are using a custom template, you'll need to update the template to correct any 
compatibility problems, and then add the compatibility declaration.
```